core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -235,16 +236,18 @@ public String toString() {
   private final List<SortOrder> sortOrders;
   private final Map<String, String> properties;
   private final long currentSnapshotId;
-  private final List<Snapshot> snapshots;
-  private final Map<Long, Snapshot> snapshotsById;
   private final Map<Integer, Schema> schemasById;
   private final Map<Integer, PartitionSpec> specsById;
   private final Map<Integer, SortOrder> sortOrdersById;
   private final List<HistoryEntry> snapshotLog;
   private final List<MetadataLogEntry> previousFiles;
-  private final Map<String, SnapshotRef> refs;
   private final List<StatisticsFile> statisticsFiles;
   private final List<MetadataUpdate> changes;
+  private final SerializableSupplier<List<Snapshot>> snapshotsSupplier;
```

Comment:
Do you think we should make this `null` after it is called, or is there value in keeping it around for debugging purposes? Setting it to null would ensure that this doesn't needlessly keep values from being garbage collected, although I can't think of anything large that would get captured in the lambda.

Reply:
I think it's worth de-referencing since there's no reason to hold the reference after.